src/main/scala/org/constellation/lb/Manager.scala

```diff
@@ -38,48 +44,80 @@ class Manager(init: NonEmptyList[Addr], config: LoadbalancerConfig)(
   private lazy val hostsRef =
     Ref.unsafe[IO, NonEmptyMap[Addr, Option[List[Info]]]](init.map(addr => addr -> None).toNem)
 
+  private lazy val sessionRef = Ref.unsafe[IO, Option[Long]](None)
+
   def node(addr: Addr)(implicit http: Client[IO]) = new RestNodeApi(addr, config.networkCredentials)
 
   private val lb = new Loadbalancer(config.port, config.`if`, config.retryAfterMinutes)
 
   def reset: IO[Unit] =
-    hostsRef
-      .modify(_ => (init.map(addr => addr -> None).toNem, ()))
-      .flatTap(_ => logger.info(s"Hosts list has been reverted to initial state."))
+    logger.info("reset") >>
+      hostsRef
+        .modify(_ => (init.map(addr => addr -> None).toNem, ()))
+        .flatMap(_ => sessionRef.set(None))
+        .flatTap(_ => logger.info(s"Hosts list and cluster session has been reverted to initial state."))
 
   def updateLbSetup(hosts: Set[Addr]): IO[Unit] =
     IO(logger.info(s"Update lb setup with hosts=$hosts"))
       .flatMap(_ => lb.withUpstream(hosts))
       .flatMap(_ => IO.unit)
 
-  private def updateProcedure(implicit client: Client[IO]) =
+  private def updateProcedure(implicit client: Client[IO]) = {
     hostsRef.get
       .flatTap(hosts => logger.info(s"Init cluster discovery from ${hosts.keys}"))
-      .flatMap(
-        hosts =>
+      .flatMap {
+        case hosts =>
           getClusterInfo(hosts.keys)(client)
-            .flatMap(
-              currentStatus =>
-                NonEmptySet
-                  .fromSet(findNewHosts(currentStatus))
-                  .map(
-                    getClusterInfo(_)
-                      .map(newHostsStatus => currentStatus ++ newHostsStatus)
-                  )
-                  .getOrElse(IO.pure(currentStatus))
-            )
-            .flatMap(
-              status =>
-                buildClusterStatus(status).flatMap {
-                  case (activeHosts, inactiveHosts) =>
-                    val clusterHosts = (activeHosts ++ inactiveHosts) // TODO: Add Eviction of inactive hosts
-                      .filterNot(addr => status.contains(addr))
-                      .foldLeft(status)((acc, addr) => acc.add(addr, Option.empty[List[Info]]))
-
-                    hostsRef.set(clusterHosts).flatTap(_ => updateLbSetup(activeHosts))
+            .flatMap { currentStatus =>
+              sessionRef.get
+                .flatMap {
+                  case oSession @ Some(_) => IO.pure(oSession)
+                  case None =>
+                    val session = findSession(filterKeys(init.toNes, currentStatus))
+                    sessionRef.set(session).map(_ => session)
```

Comment:
`.map(_ => a)` => `.as(a)`

Reply:
Any preference between 
`sessionRef.modify( _ => (session, session))` vs `sessionRef.set(session).as(session)`

neither are particularly satisfying

Reply:
if we had a generic `def dup[T](t: T): (T, T) = (t, t)` elsewhere, we can do `sessionRef.modify( _ => dup(session))`
Or even better, with `def constTuple[T](x: T)(y: T): (T, T) = (x, x)` we end with  sessionRef.modify(constTuple(session))